opm/simulators/aquifers/AquiferInterface.hpp

```diff
@@ -74,6 +76,26 @@ namespace Opm
     // Deconstructor
     virtual ~AquiferInterface() {}
 
+    void initFromRestart(const std::vector<data::AquiferData>& aquiferSoln)
+    {
+      auto xaqPos = std::find_if(aquiferSoln.begin(), aquiferSoln.end(),
+        [this](const data::AquiferData& xaq) -> bool
+      {
+        return xaq.aquiferID == this->connection_.aquiferID;
+      });
+
+      if (xaqPos == aquiferSoln.end()) {
```

Comment:
This is a tricky situation.  By definition, aquifer keywords belong to SOLUTION section and they should not be altered after simulation.  If a restart running can not find the aquifer data in the RESTART file, basically, it means some mistake has happened.  I suggest we throw here.